xds/internal/balancer/clusterresolver/clusterresolver.go

```diff
@@ -280,7 +280,7 @@ func (b *clusterResolverBalancer) handleErrorFromUpdate(err error, fromParent bo
 	// EDS resource was removed. No action needs to be taken for this, and we
 	// should continue watching the same EDS resource.
 	if fromParent && xdsresource.ErrType(err) == xdsresource.ErrorTypeResourceNotFound {
-		b.resourceWatcher.stop()
+		b.resourceWatcher.stop(false)
```

Comment:
I'm having a bit of trouble following this logic (false plumbed down). Is it to support the use case described in this comment: https://github.com/grpc/grpc-go/blob/79dd4a6fe0e6ef03dcc80877ce5d688194f1a5cc/xds/internal/balancer/clusterresolver/resource_resolver.go#L224 (i.e. reusing the lb policy in the future, so the component callback serializer the resource_resolver holds onto is still active?)

Reply:
It is basically to handle the comment here: https://github.com/grpc/grpc-go/blob/79dd4a6fe0e6ef03dcc80877ce5d688194f1a5cc/xds/internal/balancer/clusterresolver/resource_resolver.go#L246

`stop()` is called when:
- the CDS resource is deleted, or
- the LB policy is being stopped

And in the first case, when the CDS resource is added back, the cluster_resolver will get a config update with new mechanisms and will need to process it.

I initially thought about cancelling the serializer in `stop()` unconditionally and recreate it in `updateMechanisms()`, instead of in `newResourceResolver`. But this was hard to do because then you need to protect access to the serializer with a mutex, and if we have to do that, then there is no way we can guarantee that `onUpdate` won't block on the mutex.

Reply:
Ok sounds good, thanks for the explanation. It's keeping it's lifecycle coupled with this resource resolver type, which sounds fine since it's a buffer the resource resolver type uses.